components/ButtonApi.js

```diff
@@ -0,0 +1,63 @@
+import React, { useState, useEffect } from "react";
+import { View, Text, TouchableOpacity, ActivityIndicator } from "react-native";
+import styled from "styled-components/native";
+
+const Container = styled.View`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  background-color: pink;
+  padding: 10px;
+  width: 100%;
+  height: 100%;
+  text-align: center;
+`;
+
+const BoredText = styled.Text`
+  font-weight: 700;
+`;
+
+const APIButton = styled.TouchableOpacity`
+  background-color: pink;
+  justify-content: center;
+  text-align: center;
+  width: 50;
+  margin: auto;
+  margin-bottom: 2;
+`;
+
+const ButtonApi = () => {
+  const [bored, setBored] = useState({});
+  const [loading, setLoading] = useState(false);
```

Comment:
Nice and tidy! I like that you have your states together in the top, it makes it easy to understand your code!